client/basic-authorship/src/basic_authorship.rs

```diff
@@ -264,6 +285,7 @@ impl<A, B, Block, C, PR> Proposer<B, Block, C, A, PR>
 		inherent_data: InherentData,
 		inherent_digests: DigestFor<Block>,
 		deadline: time::Instant,
+		block_size_limit: Option<usize>,
```

Comment:
Yes we have one :D I actually would liked to have done this at initialization of the `ProposerFactory`. However, with parachains the limit is given by the relay chain and this limit is set in the runtime. So, this could change in any relay chain block (actually in a new session, but yeah it is dynamic)